src/server/supplements/recorderSupplement.ts

```diff
@@ -181,14 +181,16 @@ export class RecorderSupplement implements InstrumentationListener {
       recorderApp.close().catch(() => {});
     });
 
+    // Everything moved into the main world to prevent from stalling evaluation.
+
     // Input actions that potentially lead to navigation are intercepted on the page and are
     // performed by the Playwright.
     await this._context.exposeBinding('_playwrightRecorderPerformAction', false,
-        (source: BindingSource, action: actions.Action) => this._performAction(source.frame, action), 'utility');
+        (source: BindingSource, action: actions.Action) => this._performAction(source.frame, action), 'main');
```

Comment:
Running this in the main world was slow on some web sites (namely cvs.com) that the recorder was unusable. The problem was that the page included a library that overrides String.trim() with an inefficient implementation. This change will definitely regress it. Why can't controls work in the util world?

Reply:
See the bug attached.